contracts/contracts/NoLossDao.sol

```diff
@@ -110,9 +115,9 @@ contract NoLossDao is Initializable {
     _;
   }
 
-  modifier allowanceAvailable(uint256 amount) {
+  modifier allowanceAvailable(address givenAddress, uint256 amount) {
     require(
-      amount <= daiContract.allowance(msg.sender, address(this)),
+      amount <= daiContract.allowance(givenAddress, msg.sender), // checking the pool deposits contract
```

Comment:
This modifier is only used when checking balances for deposits. It will only be called by the poolDeposits contract, so it checks the given address has allowance for msg.sender (the pool deposits contract which would have initiated the function call).

Reply:
Chaning this to        address(depositContract) , more explicit and accurate